tests/system/Database/Live/WriteTypeQueryTest.php

```diff
@@ -48,11 +48,49 @@ public function testInsert(): void
 
         $this->assertTrue($this->db->isWriteType($sql));
 
-        if ($this->db->DBDriver === 'Postgre') {
-            $sql = "INSERT INTO my_table (col1, col2) VALUES ('Joe', 'Cool') RETURNING id;";
+        $sql = "WITH seqvals AS (SELECT '3' AS seqval)INSERT INTO my_table (col1, col2) SELECT 'Joe', seqval FROM seqvals;";
+
+        $this->assertTrue($this->db->isWriteType($sql));
+
+        $sql = <<<'SQL'
+                WITH seqvals AS (SELECT '3' AS seqval)
+                INSERT INTO my_table (col1, col2)
+                SELECT 'Joe', seqval
+                FROM seqvals;
+            SQL;
 
-            $this->assertFalse($this->db->isWriteType($sql));
+        $this->assertTrue($this->db->isWriteType($sql));
+
+        $assertionType = 'assertTrue';
+        if ($this->db->DBDriver === 'Postgre') {
+            $assertionType = 'assertFalse';
         }
```

Comment:
This is not acceptable.
It is too complicated, and the assertion is the opposite only on PostgreSQL.
You should write the same test (and the same result) for all DB drivers.

Reply:
Hello,

Postgres is currently the only one that is overriding the default isWriteType function to allow RETURNING; therefore, it needs to have different tests than the others.